lib/battle/CBattleInfoCallback.cpp

```diff
@@ -855,6 +855,28 @@ AccessibilityInfo CBattleInfoCallback::getAccesibility() const
 		ret[BattleHex(0, y)] = EAccessibility::SIDE_COLUMN;
 	}
 
+	//special battlefields with logically unavailable tiles
+	std::vector<BattleHex> impassableHexes;
+	switch(battleGetBattlefieldType().num)
+	{

```

Comment:
So this should be throw into some function that will provide "visibility" for programmer, and then you can change switch to if().